scipy/special/tests/test_basic.py

```diff
@@ -1545,6 +1545,36 @@ def test_errprint(self):
         assert_equal(d,b)  # makes sure state was returned
         # assert_equal(d,1-a)
 
+    def test_erf_nan_inf(self):
+        vals = [np.nan, -np.inf, np.inf]
```

Comment:
Worth something like:

```
for distribution, expected in ((special.erf, [np.nan, -1, 1]),
                                             (special.erfc, [np.nan, 2, 0]),
                                            ):
    assert_allclose(distribution(vals), expected, rtol=1e-15)
```

Suggesting only on basis that it might make it easier to add tests for all these guys at once.